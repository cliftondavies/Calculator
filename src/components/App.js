import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const { total, next, operation } = this.state;
    this.setState(calculate({ total, next, operation }, buttonName));
  }

  render() {
    const { total, next } = this.state;
    return (
      <main className="app">
        <h2>
          Let&apos;s do some math!
        </h2>

        <div>
          <Display result={next || total} />
          <ButtonPanel clickHandler={this.handleClick} />
        </div>
      </main>
    );
  }
}

export default App;
