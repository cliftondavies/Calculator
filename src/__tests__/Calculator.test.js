import renderer from 'react-test-renderer';
import App from '../components/App';

test('renders elements on calculator page', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
