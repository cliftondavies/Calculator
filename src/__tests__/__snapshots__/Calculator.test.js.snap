// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders elements on calculator page 1`] = `
<main
  className="app"
>
  <h2>
    Let's do some math!
  </h2>
  <div>
    <div
      className="display"
    >
      <span />
    </div>
    <div
      className="button-panel"
    >
      <div>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button-default"
          onClick={[Function]}
          type="button"
        >
          /
        </button>
      </div>
      <div>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button-default"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button-default"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button-default"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div>
        <button
          className="button-wide"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button-grey"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button-default"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</main>
`;
